lib/page/performMouseAction.js

```diff
@@ -30,8 +30,24 @@ const performMouseAction = async function performMouseAction(context, actionType
     return ({ status: STATUS.INVISIBLE });
   }
 
-  const overlappingNodeClassName = await context.page.evaluate(selectorNode => {
+  const overlappingNodeClassName = await context.page.evaluate(async selectorNode => {
+    const visibleRatio = await new Promise(resolve => {
+      const observer = new IntersectionObserver(entries => {
+        resolve(entries[0].intersectionRatio);
+        observer.disconnect();
+      });
+      observer.observe(selectorNode);
+    });
+
+    if (visibleRatio !== 1.0) {
+      selectorNode.scrollIntoView({
```

Comment:
I still believe that scrolling to the node when it's not requested is kinda sick. It may trigger side effects. Even those that may disturb the whole purpose of `performMouseAction`. 

But let @goganchic decide 🐨